wdl2cwl/tests/test_wdl_stdlib.py

```diff
@@ -0,0 +1,35 @@
+"""Tests for the translation of WDL Stdlib functions."""
+
+from tempfile import NamedTemporaryFile
+
+from wdl2cwl.main import convert
+
+
+def test_select_all() -> None:
+    """Tests the ``select_all`` function."""
+    # Example based off miniwdl's test_select test function:
+    # https://github.com/chanzuckerberg/miniwdl/blob/0ee2bddaea4bd2538cd21bd9b7806a6c81068b2f/tests/test_5stdlib.py#L156
+    wdl = R"""
+    version 1.0
+    task test_select_all {
+      input {
+        Int one
+        Int? two
+      }
+      command {}
+      output {
+        Array[Int] first1 = select_all([one, two])
+      }
+    }
+    """
+    with NamedTemporaryFile(mode="w", encoding="utf-8") as output:
+        output.write(wdl)
+        output.flush()
+        cwl = convert(output.name)
+        assert cwl["outputs"][0] == {
+            "id": "first1",
+            "type": {"items": "int", "type": "array"},
+            "outputBinding": {
+                "glob": "$([inputs.one, inputs.two].filter(element => element !== null) )"
```

Comment:
This doesn't make sense; why are we `glob`ing on int values?

It would be better to use standalone WDL and CWL files, so we can also test the resulting CWL files more directly.

Reply:
>This doesn't make sense; why are we globing on int values?

Does this make more sense @mr-c ?

```wdl
version 1.0
task test_select_all {
  input {
    Int one
    Int? two
  }
  command {}
  output {
    Array[Int] first1 = select_all([one, two, 1, select_first([two, one])])
  }
}
```

```cwl
class: CommandLineTool
id: test_select_all
inputs:
  - id: one
    type: int
  - id: two
    type:
      - int
      - 'null'
outputs:
  - id: first1
    type:
        items: int
        type: array
    outputBinding:
        outputEval: $([inputs.one, inputs.two, 1, [inputs.two, inputs.one].find(function(element)
            { element !== null }) ].filter(function(element) { element !== null })
            )
requirements:
  - class: InitialWorkDirRequirement
    listing:
      - entryname: script.bash
        entry: ''
  - class: InlineJavascriptRequirement
  - class: NetworkAccess
    networkAccess: true
  - class: ResourceRequirement
    outdirMin: 1024
cwlVersion: v1.2
baseCommand:
  - bash
  - script.bash
```
Looks like the string returned by the evaluation of the `WDL.Expr.Apply` is [always used](https://github.com/common-workflow-lab/wdl-cwl-translator/blob/881e646cdc552b1e496fb8833ead0df795382c47/wdl2cwl/main.py#L1111) for glob in the output.

>It would be better to use standalone WDL and CWL files, so we can also test the resulting CWL files more directly.

Will push a commit with the file and some WIP code. Marking the PR as draft :+1: 

Thanks!